diff --git a/Scripts/PlayerManager.lua b/Scripts/PlayerManager.lua
index 021a07c..942aa13 100644
--- a/Scripts/PlayerManager.lua
+++ b/Scripts/PlayerManager.lua
@@ -209,7 +209,7 @@ local function HandleTeleportPlayer(session)
               return json.stringify { error = string.format("Failed to teleport player %s: Player is inside a vehicle", playerId) }, nil, 400
             elseif pawn:IsA(vehicleClass) then
               ---@cast pawn AMTVehicle
-              PC:ServerResetVehicleAt(pawn, location, rotation, true)
+              PC:ServerResetVehicleAt(pawn, location, rotation, true, true)
             else
               error("Failed to teleport player")
             end
