diff --git a/Scripts/Webclient.lua b/Scripts/Webclient.lua
index e960356..b59cf34 100644
--- a/Scripts/Webclient.lua
+++ b/Scripts/Webclient.lua
@@ -212,14 +212,19 @@ local function RegisterEventHook(event, hookFunction, callback)
     if isEnabled and eventName then
         local status, out1, out2 = pcall(function()
             local preId, postId = RegisterHook(eventName, function(self, ...)
-                local status, result = pcall(hookFunction, self, ...)
-                if status then
-                    if result then
-                        CreateEventWebhook(eventName, result, callback)
-                    end
-                else
-                    LogOutput("ERROR", "Failed to execute event hook: %s", result)
-                end
+                LogOutput("INFO", "Event hook start: %s", eventName)
+                local params = {...}
+                ExecuteAsync(function()
+                  local status, result = pcall(hookFunction, self, table.unpack(params))
+                  if status then
+                      if result then
+                          CreateEventWebhook(eventName, result, callback)
+                          LogOutput("INFO", "Event hook created: %s", eventName)
+                      end
+                  else
+                      LogOutput("ERROR", "Failed to execute event hook: %s", result)
+                  end
+                end)
             end)
             return preId, postId
         end)
